src/main/java/com/opentok/OpenTok.java

```diff
@@ -33,427 +33,506 @@
-            throw new RequestException("Exception mapping json: " + e.getMessage());
-        } catch (IOException e) {
-            throw new RequestException("Exception mapping json: " + e.getMessage());
-        }
-    }
-
-    /**
-     * Starts archiving an OpenTok session. This version of the <code>startArchive()</code> method
-     * lets you disable audio or video recording.
-     * <p>
-     * Clients must be actively connected to the OpenTok session for you to successfully start
-     * recording an archive.
-     * <p>
-     * You can only record one archive at a time for a given session. You can only record archives
-     * of sessions that use the OpenTok Media Router (sessions with the
-     * <a href="http://tokbox.com/opentok/tutorials/create-session/#media-mode">media mode</a>
-     * set to routed); you cannot archive sessions with the media mode set to relayed.
-     * <p>
-     * For more information on archiving, see the
-     * <a href="https://tokbox.com/opentok/tutorials/archiving/">OpenTok archiving</a>
-     * programming guide.
-     * 
-     * @param sessionId The session ID of the OpenTok session to archive.
-     * 
-     * @param properties This ArchiveProperties object defines options for the archive.
-     *
-     * @return The Archive object. This object includes properties defining the archive, including the archive ID.
-     */
-    public Archive startArchive(String sessionId, ArchiveProperties properties) throws OpenTokException {
-        if (sessionId == null || sessionId == "") {
-            throw new InvalidArgumentException("Session not valid");
-        }
-        // TODO: do validation on sessionId and name
-        String archive = this.client.startArchive(sessionId, properties);
-        try {
-            return archiveReader.readValue(archive);
-        } catch (Exception e) {
-            throw new RequestException("Exception mapping json: " + e.getMessage());
-        }
-    }
+	private int apiKey;
+	private String apiSecret;
+
+	private HttpClient client;
+	static protected ObjectReader archiveReader = new ObjectMapper().reader(Archive.class);
+	static protected ObjectReader archiveListReader = new ObjectMapper().reader(ArchiveList.class);
+	static final String defaultApiUrl = "https://api.opentok.com";
+
+	/**
+	 * Creates an OpenTok object.
+	 *
+	 * @param apiKey
+	 *            Your OpenTok API key. (See the
+	 *            <a href="https://dashboard.tokbox.com">OpenTok dashboard</a>
+	 *            page.)
+	 * @param apiSecret
+	 *            Your OpenTok API secret. (See the
+	 *            <a href="https://dashboard.tokbox.com">OpenTok dashboard</a>
+	 *            page.)
+	 */
+	public OpenTok(int apiKey, String apiSecret) {
+		this.apiKey = apiKey;
+		this.apiSecret = apiSecret;
+
+		HttpClient.Builder clientBuilder = new HttpClient.Builder(apiKey, apiSecret).apiUrl(this.defaultApiUrl);
+
+		this.client = clientBuilder.build();
+	}
+
+	public void setClient(HttpClient client) {
```

Comment:
New public method? You cannot do this without some API review.`HttpClient` is a utility class that's not supposed to have any exposure in this interface.